enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -5980,6 +5980,456 @@ class AdjointGenerator
+                     Builder2),
+              lookup(gutils->getNewFromOriginal(call.getArgOperand(5)),
+                     Builder2),
+              lookup(gutils->getNewFromOriginal(call.getArgOperand(4)),
+                     Builder2),
+              lookup(gutils->getNewFromOriginal(call.getArgOperand(6)),
+                     Builder2),
+              gutils->invertPointerM(call.getArgOperand(12), Builder2),
+              salda,
+              structarg2,
+              saldb,
+              doneval,
+              gutils->invertPointerM(call.getArgOperand(7), Builder2),
+              saldc};
+          safunccall = Builder2.CreateCall(dfunc, safuncargs);
+        }
+        if (bactive) {
+          if (cast<ConstantInt>(call.getArgOperand(1))->getValue() == 112 ||
+              cast<ConstantInt>(call.getArgOperand(1))->getValue() == 113) {
+            sbatrans = Builder2.getInt32(111);
+            if (cast<ConstantInt>(call.getArgOperand(0))->getValue() == 102)
+              sblda = lookup(gutils->getNewFromOriginal(call.getArgOperand(5)),
+                             Builder2);
+            else
+              sblda = lookup(gutils->getNewFromOriginal(call.getArgOperand(3)),
+                             Builder2);
+          } else if (cast<ConstantInt>(call.getArgOperand(1))->getValue() ==
+                     111) {
+            sbatrans = Builder2.getInt32(112);
+            if (cast<ConstantInt>(call.getArgOperand(0))->getValue() == 102)
+              sblda = lookup(gutils->getNewFromOriginal(call.getArgOperand(3)),
+                             Builder2);
+            else
+              sblda = lookup(gutils->getNewFromOriginal(call.getArgOperand(5)),
+                             Builder2);
+          } else
+            assert(false && "Wrong value");
+          SmallVector<Value *, 13> sbfuncargs = {
+              lookup(gutils->getNewFromOriginal(call.getArgOperand(0)),
+                     Builder2),
+              sbatrans,
+              Builder2.getInt32(111),
+              lookup(gutils->getNewFromOriginal(call.getArgOperand(5)),
+                     Builder2),
+              lookup(gutils->getNewFromOriginal(call.getArgOperand(4)),
+                     Builder2),
+              lookup(gutils->getNewFromOriginal(call.getArgOperand(3)),
+                     Builder2),
+              lookup(gutils->getNewFromOriginal(call.getArgOperand(6)),
+                     Builder2),
+              structarg1,
+              sblda,
+              gutils->invertPointerM(call.getArgOperand(12), Builder2),
+              sbldb,
+              doneval,
+              gutils->invertPointerM(call.getArgOperand(9), Builder2),
+              sbldc};
+          sbfunccall = Builder2.CreateCall(dfunc, sbfuncargs);
+        }
+        auto scfunc = gutils->oldFunc->getParent()->getOrInsertFunction(
+            scfuncname, Builder2.getVoidTy(), Builder2.getInt32Ty(),
```

Comment:
Are we always sure this is an int32? Can we instead use whatever the type of the constantint is (if, for some reason, it is say an int64, etc).